diff --git a/includes/menu.inc b/includes/menu.inc
index 128a576..87b976e 100644
--- a/includes/menu.inc
+++ b/includes/menu.inc
@@ -1,18 +1,20 @@
 <?php
+
 /**
  * @file
  * Menu functions and hooks for the D8 caching system backport.
  */
 
-// @todo In theory we should take over cache_menu and re-act to cache clear
-//       events by invalidating a menu_view tag.
+/* @todo In theory we should take over cache_menu and re-act to cache clear
+ * events by invalidating a menu_view tag.
+ */
 
 /**
  * Implements hook_preprocess_menu_link().
  */
 function d8cache_preprocess_menu_link(&$variables) {
   $link = $variables['element']['#original_link'];
-  
+
   // Unfortunately preprocess_menu_tree does not have the information we need,
   // so we need to add the cache tag for each menu link in a menu separately.
   drupal_add_cache_tags(_d8cache_menu_link_get_cache_tags($link));
@@ -23,14 +25,14 @@ function d8cache_preprocess_menu_link(&$variables) {
  */
 function d8cache_preprocess_links(&$variables) {
   $tag_map = array();
-
-  foreach ($variables['links'] as $link) {
-    if (isset($link['#menu_name'])) {
-      $tag_map['menu:' . strtr($link['#menu_name'], '-', '_')] = TRUE;
+  if (isset($variables['links'])) {
+    foreach ($variables['links'] as $link) {
+      if (isset($link['#menu_name'])) {
+        $tag_map['menu:' . strtr($link['#menu_name'], '-', '_')] = TRUE;
+      }
     }
+    drupal_add_cache_tags(array_keys($tag_map));
   }
-
-  drupal_add_cache_tags(array_keys($tag_map));
 }
 
 /**
@@ -62,15 +64,18 @@ function d8cache_menu_link_delete($link) {
   drupal_invalidate_cache_tags(_d8cache_menu_link_get_cache_tags($link));
 }
 
-// -----------------------------------------------------------------------
-// Contrib hooks
-
-// -----------------------------------------------------------------------
-// Public API
-
-// -----------------------------------------------------------------------
-// Helper functions
+/**
+ * Contrib hooks.
+ *
+ * -----------------------------------------------------------------------
+ * Public API.
+ * -----------------------------------------------------------------------
+ * Helper functions.
+ */
 
+/**
+ * Helper function to transform machine names.
+ */
 function _d8cache_menu_link_get_cache_tags($link) {
   return array(
     'menu:' . strtr($link['menu_name'], '-', '_'),
